beacon_node/beacon_chain/src/migrate.rs

```diff
@@ -216,6 +211,11 @@ impl<E: EthSpec, Hot: ItemStore<E>, Cold: ItemStore<E>> Migrate<E, Hot, Cold>
         ) {
             eprintln!("Pruning error: {:?}", e);
         }
+
+        if let Err(e) = process_finalization(self.db.clone(), state_root, &new_finalized_state) {
```

Comment:
The same change in ordering needs to happen in the `BackgroundMigrator` as well

Reply:
That's a very good point